### PR TITLE
Get rid of empty init functions

### PR DESCRIPTION
**Pull request by Frotty**
Removes empty package init functions, see https://github.com/wurstscript/WurstScript/issues/511

### CODE REVIEW

File: de.peeeq.wurstscript/src/main/java/de/peeeq/wurstscript/translation/imoptimizer/InitFunctionCleaner.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+public class InitFunctionCleaner {
+
+    public static void clean(ImProg prog) {
+        List<ImFunction> initFuncs = prog.getFunctions().stream().filter(func -> func.getName().startsWith
+                ("init_") && func.getBody().size() == 1).collect(Collectors.toList());
+
+        initFuncs.forEach(func -> {
+            ImVar globalBridge = null;
+            for (ImVar global : prog.getGlobals()) {
+                if (global.getName().equals("ref_function_" + func.getName())) {
```
Comment (peq) [inline de.peeeq.wurstscript/src/main/java/de/peeeq/wurstscript/translation/imoptimizer/InitFunctionCleaner.java]: I don't like using the name here. Names should not matter in the intermediate language.

File: de.peeeq.wurstscript/src/main/java/de/peeeq/wurstscript/translation/imoptimizer/InitFunctionCleaner.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class InitFunctionCleaner {
+
+    public static void clean(ImProg prog) {
+        List<ImFunction> initFuncs = prog.getFunctions().stream().filter(func -> func.getName().startsWith
+                ("init_") && func.getBody().size() == 1).collect(Collectors.toList());
+
+        initFuncs.forEach(func -> {
```
Comment (peq) [inline de.peeeq.wurstscript/src/main/java/de/peeeq/wurstscript/translation/imoptimizer/InitFunctionCleaner.java]: please use normal for-loops -- better performance, better stack traces, better debugging, easier to read

File: de.peeeq.wurstscript/src/main/java/de/peeeq/wurstscript/translation/imoptimizer/InitFunctionCleaner.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                                    int finalI = i;
+                                    ((ImFunctionCall) imStmt).getArguments().forEach(arg -> {
+                                        if ((arg instanceof ImFunctionCall) && ((ImFunctionCall) arg).getFunc().getName().equals("Condition")) {
+                                            ImExpr next = ((ImFunctionCall) arg).getArguments().iterator().next();
+                                            if ((next instanceof ImVarAccess) && ((ImVarAccess) next).getVar().structuralEquals(var)) {
+                                                imStmt.replaceBy(JassIm.ImNull());
+                                                if (body.get(finalI - 1) instanceof ImFunctionCall) {
+                                                    body.get(finalI - 1).replaceBy(JassIm.ImNull());
+                                                }
+                                                body.get(finalI + 1).replaceBy(JassIm.ImNull());
```
Comment (peq) [inline de.peeeq.wurstscript/src/main/java/de/peeeq/wurstscript/translation/imoptimizer/InitFunctionCleaner.java]: What is the -1 and +1 doing here? Looks like you are assuming that there are always certain other statements around a call to `TriggerAddCondition` but that does not seem very reliable to me.

File: de.peeeq.wurstscript/src/main/java/de/peeeq/wurstscript/translation/imoptimizer/InitFunctionCleaner.java (hunk 1 above)
Comment (Frotty) [reply de.peeeq.wurstscript/src/main/java/de/peeeq/wurstscript/translation/imoptimizer/InitFunctionCleaner.java]: What to use then?

File: de.peeeq.wurstscript/src/main/java/de/peeeq/wurstscript/translation/imoptimizer/InitFunctionCleaner.java (hunk 3 above)
Comment (Frotty) [reply de.peeeq.wurstscript/src/main/java/de/peeeq/wurstscript/translation/imoptimizer/InitFunctionCleaner.java]: The init calls are always generated the same way. Unless someone names a function "ref_function_" or so (which is the name issue above) then it seems reliable to me.

**Comment by Frotty** [comment]
> Would it not be easier to improve the generation of the bridge-functions instead of trying to find and optimize them after the fact?

No idea where to start with that tbh. The old ticket & solution have been around for a while, this is one without inlining again.

**Comment by Frotty** [comment]
@peq ?

**Pull request closed without merging**